### ISSUE TITLE
Use Gauss points on some mortars

### ISSUE THREAD

**Issue by nilsvu** [sxs-collaboration/spectre#871]
# Feature request:

We may want to use Gauss points for a mortar when both elements have to interpolate to it anyway. That situation occurs during mesh-refinement when the larger element has a higher resolution than the smaller element. It basically just involves changing `dg::mortar_mesh` to use `Spectral::Quadrature::Gauss` in that situation, and hoping this doesn't break anything (which is a good test in itself).

The reason we want to do this is that we gain accuracy on the mortar for free, by performing quadratures on Gauss points, instead of Gauss-Lobatto points.

### Component:

- [x] Code

**Comment by teukolsky** [comment]
Is this something that would be useful for the elliptic solver? I don't think it's necessary for evolutions, since one interface is always just a copy and we can handle the accuracy issue via projection, which is no more expensive than anything else (just a matrix-vector multiply).

**Comment by nilsvu** [comment]
@teukolsky this is what @trevor-vincent does in https://github.com/trevor-vincent/d4est (I think). It's probably not necessary, but could enhance performance for free.

You say one interface is always just a copy, in which case my statement would be false. However, as I understand it there is a case where both elements have to interpolate to the mortar, which is the one I describe above. It occurs when the element that is covered only half by the mortar because it's less h-refined has a higher p-resolution than then smaller element which is covered in full by the mortar. In that case we want to use the higher p-resolution from the larger element for the mortar. This means the points from the larger element don't match the mortar because of the size difference, and the points from the smaller element don't match because of the p-resolution difference. Both have to interpolate and may as well do that to Gauss points instead of Gauss-Lobatto points.

Am I getting something wrong here?

**Comment by teukolsky** [comment]
Yes, you are correct that there is a case when both elements have to interpolate to the mortar. However, I don't think it's ever necessary to use Gauss points to obtain full quadrature accuracy - there are alternative simpler methods as described in my projection notes (which will hopefully be an part of one of our papers, or be published separately).

The projection from the small element to the mortar is exactly the same in the h-refinement case as it is for p-refinement, since the mortar is the same size. We know how to do that to full accuracy without Gauss points. For the large element, going from the element to the mortar one can show that projection is the same as interpolation, even with the Lobatto points. For the projection from the mortar back to the element, working in coordinate space does require Gauss points for full accuracy. However, I derived a matrix one can use in spectral space that is exact for the Lobatto points. The matrix can be pre-computed, and reduces the projection to a matrix-vector multiply like all the other cases.
